riscv/csrs.cc

```diff
@@ -510,6 +510,7 @@ reg_t mstatus_csr_t::compute_mstatus_initial_value() const noexcept {
                               | (proc->extension_enabled_const('S') ? MSTATUS_SBE : 0)
                               | MSTATUS_MBE;
   return 0
+         | (proc->extension_enabled_const('U') ? PRV_U : PRV_M)
```

Comment:
Don't you need to make this something like

    (proc->extension_enabled_const('U') ? 0 : set_field((reg_t)0, MSTATUS_MPP, PRV_M))

or

    set_field((reg_t)0, MSTATUS_MPP, proc->extension_enabled_const('U') ? PRV_U : PRV_M)

so the bits get shifted into the right position within `mstatus`?

Reply:
@aswaterman Thank you for the review. Yes, we need to shift into the proper position. Should we simplify the source code by letting the MPP always initialize to M-mode?
```| MSTATUS_MPP # Initialize MPP to M-mode```

Reply:
Although I agree that would be valid I would prefer to avoid changing the behavior for the case that U-mode is present.